### ISSUE TITLE
feat: Research and implement web-based visualization for database content

### ISSUE THREAD

**Issue by prawwtocol** [manytask/manytask#198]
### Is there an existing planned feature for this?

- [X] I have searched the existing issues, it is original

### Type of feature

`feat` - a new feature

### Current behavior

Currently, our web interface contains links that redirect users to Google Sheets. As we're migrating from Google Sheets to a database solution, we need to implement a proper way to visualize database content directly in our web interface.

Current behavior:
- Users click a link 
- that opens Google Sheets

### Suggested solution

Desired behavior:
- Users click a link that opens a table view within our web interface
- Data is fetched from our database and displayed [in a table format]
- [Optional] The table should be user-friendly and support basic functionality (sorting, filtering, etc.)

Research tasks:
1. Investigate available solutions for web-based DB visualization:
2. Compare solutions based on:
   - Feature set (sorting, filtering, pagination)
   - Browser compatibility
   - License and pricing (if applicable)

Implementation considerations:
- Mobile responsiveness
- Loading states
- Error handling
- Data refresh mechanism

### Terms and Conditions

- [X] I agree to follow this project's Code of Conduct.

**Comment by feldsherov** [comment]
My plan here was the following:

Now student and lecturer can open link and see google sheet with all students/tasks/scores.
We need to repeat the same, but in manytask ui.

In terms of API, I would expose just a handler which by a given course will return to you:
- all tasks in course
- all students in course
- all scores in course

It may worth of spiting handler into more granular, but I wouldn't over-complicate now.

On the client side result of this handler is fed into any JS library, which can render a table.
Format of the table should be mostly similar to what we have in gdoc now.

**Comment by prawwtocol** [comment]
@feldsherov 

I agree with this. Something similar to the google sheets read-only spreadsheet interface should be okay IMO.
